pkg/plugin/plugin.go

```diff
@@ -1,12 +1,92 @@
 package plugin
 
 import (
+	"fmt"
+	"log"
+	"net/http"
 	"os"
 
+	"github.com/gin-gonic/gin"
+
 	"github.com/vincenzopalazzo/cln4go/plugin"
 )
 
-type PluginState struct{}
+type PluginState struct {
+	Server   *gin.Engine
+	Password string
+}
+
+type Login struct {
+	Password string `json:"password" binding:"required"`
+}
+
+type TLSFiles struct {
+	Ca        string `json:"ca"`
+	ClientKey string `json:"client_key"`
+	Client    string `json:"client"`
+}
+
+func readTlsFile() (TLSFiles, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return TLSFiles{}, err
+	}
+	clientKey, err := os.ReadFile(home + "/.lightning/testnet/client-key.pem")
+	if err != nil {
+		return TLSFiles{}, err
+	}
+	ca, err := os.ReadFile(home + ".lightning/testnet/ca.pem")
+	if err != nil {
+		return TLSFiles{}, err
+	}
+	client, err := os.ReadFile(home + ".lightning/testnet/client.pem")
+	if err != nil {
+		return TLSFiles{}, err
+	}
+	files := TLSFiles{Ca: string(ca), ClientKey: string(clientKey), Client: string(client)}
+	return files, nil
+}
+
+type StartServer[T PluginState] struct{}
+
+func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]string) (map[string]string, error) {
```

Comment:
```suggestion
func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]any) (map[string]any, error) {
```

This is the bug, it is not in the cln4go lib